packages/ove-asset-manager/src/OVE.Service.AssetManager/Domain/S3AssetFileOperations.cs

```diff
@@ -0,0 +1,179 @@
+                    UseHttp = true, 
+                    ForcePathStyle = true
+                }
+            );
+            _logger.LogInformation("Created new S3 Client");
+            return s3Client;
+        }
+        
+        #region Implementation of IFileOperations
+
+        public string ResolveFileUrl(OVEAssetModel asset) {
+            var url = _configuration.GetValue<string>(S3ClientServiceUrl)
+                      + asset.Project + "/" + asset.StorageLocation;
+            return url;
+        }
+
+#pragma warning disable 1998
+        public async Task<bool> MoveFile(OVEAssetModel oldAsset, OVEAssetModel newAsset) {
+#pragma warning restore 1998
+            //todo this is hard because we might have to move between s3 buckets and that is complex
+            // https://stackoverflow.com/questions/9664904/best-way-to-move-files-between-s3-buckets
+            throw new NotImplementedException();
+        }
+
+        public async Task<bool> DeleteFile(OVEAssetModel asset) {
+            _logger.LogInformation("about to delete file "+asset);
+            try {
+                using (var s3Client = GetS3Client(_configuration)) {
+                    // delete folder containing the file and everything
+                    
+                    ListObjectsResponse files = null;
+                    while (files == null || files.S3Objects.Any()) {
+                        if (files != null && files.S3Objects.Any()) {
+                            foreach (var o in files.S3Objects) {
+                                await s3Client.DeleteObjectAsync(asset.Project, o.Key);
+                            }
+                        }
+
+                        // find more files
+                        files = await s3Client.ListObjectsAsync(new ListObjectsRequest {
+                            BucketName = asset.Project,
+                            Prefix = asset.GetStorageGuid()
+                        });
+
+                    }
+
+                    // if the bucket is empty then delete it 
+                    var res = await s3Client.ListObjectsAsync(asset.Project);
+                    if (!res.S3Objects.Any()) {
+                        await s3Client.DeleteBucketAsync(asset.Project);
+                    }
+                }
+
+                _logger.LogInformation("deleted file on s3 correctly");
+                return true;
+            }
+            catch (Exception e) {
+                _logger.LogError(e, "Failed to delete an s3 file for " + asset);
+                return false;
+            }
+        }
+
+        public async Task<bool> SaveFile(OVEAssetModel asset, IFormFile upload) {
+            _logger.LogInformation("about to upload "+asset);
+
+            // set up the filename            
+            asset.StorageLocation = Guid.NewGuid() +"/"+S3Santize(Path.GetFileNameWithoutExtension(upload.FileName))+Path.GetExtension(upload.FileName).Substring(0,4).ToLower();

```

Comment:
So what happens if I upload a file.graphml ?
Is my Storage Location not then file.grap? (or on testing, .gra)